packages/core/src/connectors/wechat/index.ts

```diff
@@ -0,0 +1,131 @@
+export const metadata: ConnectorMetadata = {
+  id: 'wechat',
+  type: ConnectorType.Social,
+  name: {
+    en: 'Sign In with WeChat',
+    'zh-CN': 'WeChat 登录',
+  },
+  // TODO: add the real logo URL (LOG-1823)
+  logo: './logo.png',
+  description: {
+    en: 'Sign In with WeChat',
+    'zh-CN': 'WeChat 登录',
+  },
+  readme: existsSync(pathToReadmeFile)
+    ? readFileSync(pathToReadmeFile, 'utf8')
+    : readmeContentFallback,
+};
+
+const weChatConfigGuard = z.object({ appId: z.string(), appSecret: z.string() });
+
+type WeChatConfig = z.infer<typeof weChatConfigGuard>;
+
+export const validateConfig: ValidateConfig = async (config: unknown) => {
+  const result = weChatConfigGuard.safeParse(config);
+
+  if (!result.success) {
+    throw new ConnectorError(ConnectorErrorCodes.InvalidConfig, result.error.message);
+  }
+};
+
+export const getAuthorizationUri: GetAuthorizationUri = async (redirectUri, state) => {
+  const { appId } = await getConnectorConfig<WeChatConfig>(metadata.id);
+
+  return `${authorizationEndpoint}?${stringify({
+    appid: appId,
+    redirect_uri: encodeURI(redirectUri), // The variable `redirectUri` should match {appId, appSecret}
+    response_type: 'code',
+    scope,
+    state,
+  })}`;
+};
+
+export const getAccessToken: GetAccessToken = async (code) => {
+  type AccessTokenResponse = {
+    access_token: string;
+    openid: string;
+    expires_in: number; // In seconds
+    refresh_token: string;
+    scope: string;
+  };
+
+  const config = await getConnectorConfig<WeChatConfig>(metadata.id);
+  const { appId: appid, appSecret: secret } = config;
+
+  const { access_token: accessToken, openid } = await got
+    .get({
+      url: accessTokenEndpoint,
+      searchParams: { appid, secret, code, grant_type: 'authorization_code' },
+      timeout: await getConnectorRequestTimeout(),
+    })
+    .json<AccessTokenResponse>();
+
+  assertThat(accessToken, new ConnectorError(ConnectorErrorCodes.SocialAuthCodeInvalid));
+
+  return { accessToken, openid };
+};
+
+export const getUserInfo: GetUserInfo = async (accessTokenObject) => {
+  type UserInfoResponse = {
+    unionid?: string;
+    headimgurl?: string;
+    nickname?: string;
+  };
+
+  const { accessToken, openid } = accessTokenObject;
+
+  try {
+    const { unionid, headimgurl, nickname } = await got
+      .get({
+        url: userInfoEndpoint,
+        searchParams: { access_token: accessToken, openid },
+        timeout: await getConnectorRequestTimeout(),
+      })
+      .json<UserInfoResponse>();
+
+    const userId = unionid ?? openid;
+
+    if (!userId) {
+      throw new ConnectorError(ConnectorErrorCodes.General);
+    }
+
+    return { id: userId, avatar: headimgurl, name: nickname };
+  } catch (error: unknown) {
+    if (error instanceof GotRequestError && error.response?.statusCode === 401) {
+      throw new ConnectorError(ConnectorErrorCodes.SocialAccessTokenInvalid);
```

Comment:
Just a reminder: make sure to test this case in real env.

Reply:
Thanks for the mention :P